aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/service/AWSS3StorageService.kt

```diff
@@ -182,12 +184,15 @@ internal class AWSS3StorageService(
      * @param path The path to list items from
      * @return A list of parsed items
      */
-    override fun listFiles(path: String, prefix: String): MutableList<StorageItem>? {
-        val items = mutableListOf<StorageItem>()
-        runBlocking {
+    override fun listFiles(path: String, prefix: String, subPathStrategy: SubpathStrategy?): StorageListResult {
+        return runBlocking {
```

Comment:
There should not be `runBlocking` usage in production code 🤦 I can see that it's always run on an executor from the calling class so it's not going to cause problems, but this is still a coroutine anti-pattern. Out of scope to fix this in this PR, but we need to revisit the design of this plugin.

Reply:
Sounds good, I'll take a note and look and cleaning up the plugin.